cross/c2000.txt

```diff
@@ -0,0 +1,26 @@
+# This file assumes that path to the Texas Instruments C20000 toolchain is added
+# to the environment(PATH) variable, so that Meson can find
+# cl2000 and ar2000 while building.
+[binaries]
+c = 'cl2000'
+ar = 'ar2000'
+strip = 'cl2000'
+
+[host_machine]
+system = 'bare metal'
+cpu_family = 'c2000'
+cpu = 'c28x'
```

Comment:
Aren't `cpu` and `cpu_family` the wrong way around here?

The convention we have is to write `baremetal` as one word, rather than splitting it with a space.

Both of these issues also apply to the other cross file listed below.

Reply:
cpu_family and cpu are that way round in all the other cross files.
And "bare metal" has 3 other cross files, "baremetal" only has 1 cross-file ?

Reply:
I read through TI's spec sheets and the naming is very confusing. But at least in the other cross file `dspic` should be the family and based on googling `xc16` is the toolchain name rather than any processor code. Based on the contents, maybe its values should be `dspic` for the cpu_family and `33EP64MC203` for the cpu?

Can you post a good link to a page that lists the naming conventions for these, I can't seem to be able to find one?

Regardless, the values of `cpu_family` in both these cross files should be found in the reference table.

The baremetal thing we can fix with a harmonizing commit afterwards. It does not need to be done here.

Reply:
the cpu family for c2000 is definitely c2000. Here http://www.ti.com/microcontrollers/overview.html it offers 3 options, arm, msp430 and c2000

the cpu subgroup is c28x. the exact  #microcontroller part number is the TMS320F280049C, they have thousands of parts.

Which way would you prefer?
cpufamily=c2000 and cpu=c28x both documented in the reference table?
or
cpufamily=c2000 documented in the reference table. and cpu=TMS320F280049C not documented in the reference table because there are 1000's of them

For the xc16 one, they simply refer to there "16-bit PIC® Microcontrollers Family" xc16 is easier to understand since its the toolchain name?
https://www.microchip.com/design-centers/16-bit

They have CPU 16 bit subgroups, PIC24F, PIC24H, PIC24E, dsPIC30, dsPIC33F, dsPIC33E, dsPIC33C etc..
so cpufamily=xc16 and cpu=dspic kinda works?

Reply:
`cpu` is not documented in any table, we specifically say that its value can be pretty much anything. Based on the link, `c2000` seems exactly like a suitable value for `cpu_family`.

The other one is trickier. Maybe it should be `pic` for `cpu_family` and then the people can put the specific model in `cpu` as they see fit? Or, alternatively, the family values could be `pic16`, `pic24`, `dspic` et al? This would better match the idea of mapping processors (roughly) to their instruction sets?

Reply:
`pic` by itself won't work because they have a separate pic 8bit and a 32bit compiler that you need to fit in.

so yes I will change the  `cpu_family` table to include `pic24` and `dspic`. Note the `pic16` is an 8bit micro and uses the other xc8 compiler.